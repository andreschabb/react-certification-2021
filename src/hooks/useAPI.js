import { useEffect, useState } from 'react';

import youtube from '../apis/youtube';
import { DEFAULT_SEARCH } from '../utils/constants';

const useAPI = (search = DEFAULT_SEARCH) => {
  const [videos, setVideos] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);

  useEffect(() => {
    const onSearch = async (searchedText) => {
      try {
        setLoading(true);
        const response = await youtube.get('/search', {
          params: {
            q: searchedText,
          },
        });
        setVideos(response.data.items);
      } catch (e) {
        setError(e);
      } finally {
        setLoading(false);
      }
    };
    onSearch(search);
  }, [search]);

  return [videos, loading, error];
};

export default useAPI;
