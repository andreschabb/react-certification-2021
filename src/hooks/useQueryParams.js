import { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';

const useQueryParams = () => {
  const [searchedText, setSearchedText] = useState('');
  const [videoId, setVideoId] = useState('');
  const location = useLocation();

  useEffect(() => {
    const query = new URLSearchParams(location.search);
    setSearchedText(query.get('searchedText'));
    setVideoId(query.get('id'));
  }, [location]);

  return { searchedText, videoId };
};

export default useQueryParams;
