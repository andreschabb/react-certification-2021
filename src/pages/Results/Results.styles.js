import styled from 'styled-components';

export const Container = styled.section`
  text-align: center;
  height: 100%;
  width: 80%;
  margin-bottom: 5%;
`;

export const Loader = styled.section`
  height: 100vh;
`;

export const Title = styled.h1`
  font-size: 3rem;
  letter-spacing: -6px;
  font-family: 'Lucida Console', 'Courier New', monospace;
  font-weight: normal;
  color: ${({ isThemeLight }) => (isThemeLight ? 'black' : 'white')};

  @media screen and (max-width: 750px) {
    font-size: 2rem;
    letter-spacing: -4px;
  }

  @media screen and (max-width: 500px) {
    font-size: 1.5rem;
    letter-spacing: -2px;
  }
`;
