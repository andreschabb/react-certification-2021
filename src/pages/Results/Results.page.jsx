import React from 'react';

import VideoList from '../../components/VideoList';
import useAPI from '../../hooks/useAPI';
import useQueryParams from '../../hooks/useQueryParams';
import { useGlobalState } from '../../providers/GlobalState/Provider';
import { Container, Title, Loader } from './Results.styles';

const ResultsPage = () => {
  const { searchedText } = useQueryParams();
  const [videos, loading] = useAPI(searchedText);
  const { state } = useGlobalState();
  const { isThemeLight } = state;

  return !loading ? (
    <Container>
      <Title isThemeLight={isThemeLight} data-testid="results-message">
        Results for {`"${searchedText}"`}:{' '}
      </Title>
      <VideoList videos={videos} />
    </Container>
  ) : (
    <Loader />
  );
};

export default ResultsPage;
