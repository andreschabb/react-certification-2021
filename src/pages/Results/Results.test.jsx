import React from 'react';
import { render, screen } from '@testing-library/react';

import useAPI from '../../hooks/useAPI';
import useQueryParams from '../../hooks/useQueryParams';
import Results from './Results.page';

jest.mock('../../hooks/useAPI');
jest.mock('../../hooks/useQueryParams');

const mockedVideos = [
  {
    id: { kind: 'youtube#video', videoId: 'nmXMgqjQzls' },
    snippet: {
      title: 'Video Tour | Welcome to Wizeline Guadalajara',
      description:
        'Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...',
      thumbnails: {
        medium: {
          url: 'https://i.ytimg.com/vi/nmXMgqjQzls/medium.jpg',
          width: 120,
          height: 90,
        },
      },
    },
  },
  {
    id: { kind: 'youtube#video', videoId: 'nmXMgqjQlhj' },
    snippet: {
      title: 'Title',
      description: 'Description',
      thumbnails: {
        medium: {
          url: 'https://i.ytimg.com/vi/nmXMgqjQzls/medium.jpg',
          width: 120,
          height: 90,
        },
      },
    },
  },
];

describe('Results', () => {
  it('should renders without videos', () => {
    useAPI.mockImplementation(() => [[], false]);
    useQueryParams.mockImplementation(() => ({ searchedText: 'Happy dogs' }));

    const { container } = render(<Results />);

    expect(screen.getByTestId('results-message')).toBeTruthy();

    expect(container).toMatchSnapshot();
  });

  it('should renders with videos', () => {
    useAPI.mockImplementation(() => [mockedVideos, false]);
    useQueryParams.mockImplementation(() => ({ searchedText: 'Happy dogs' }));
    render(<Results />);

    expect(screen.getByTestId('results-message')).toBeTruthy();

    expect(screen.getByTestId('video-list')).toBeTruthy();
    expect(screen.getAllByTestId('video-item').length).toBe(2);
  });
});
