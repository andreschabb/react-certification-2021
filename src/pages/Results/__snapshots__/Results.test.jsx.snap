// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should renders without videos 1`] = `
<div>
  <section
    class="sc-iBPRYJ htirTG"
  >
    <h1
      class="sc-pFZIQ dEVQfo"
      data-testid="results-message"
    >
      Results for 
      "Happy dogs"
      :
       
    </h1>
    <div
      class="sc-gKsewC ikKpsV"
      data-testid="video-list"
    />
  </section>
</div>
`;
